Add Openstack support

cestef opened the pull request:
This pretty much adds support for OS's ["Swift"](https://docs.openstack.org/swift/latest/index.html) service (Object Storage), feel free to reach me out for any modifications etc.

cestef committed:
Add Openstack support

cestef committed:
Merge branch 'trunk' of https://github.com/cestef/zipline into trunk

cestef committed:
Fix datasource getting

cestef committed:
Restore example config + remove useless vscode dir

cestef committed:
Restore example config + remove useless vscode dir

cestef committed:
Add config.ts to the entryPoints list

diced reviewed (changes requested):
Could you add the new configuration changes to the `src/lib/config/readConfig.ts` file? And for the Openstack.ts where you import the logger use `lib/logger` instead of `../logger.

cestef committed:
Fix indenting problems with switch-case

cestef committed:
Replace openstack-swift-client

cestef committed:
Add openstack to the config validator

cestef commented:
Should be all good now ! Removed the weird library and also added openstack to the config validation part (switched from endless `if`s and `else if`s to a regular `switch/case`)

cestef commented:
Oh yeah and I also removed a weird ESlint rule that was forcing me to indent the switch cases weirdly, don't know if that was intentional, feel free to remove it

cestef committed:
Rename Openstack to Swift

cestef committed:
Better error handling for swift

cestef committed:
More error handling

diced committed:
Update Swift.ts
